distributed/comm/tcp.py

```diff
@@ -384,21 +385,37 @@ def _check_encryption(self, address, connection_args):
 
 
 class BaseTCPConnector(Connector, RequireEncryptionMixin):
-    _executor = ThreadPoolExecutor(2, thread_name_prefix="TCP-Executor")
+    _executor: ClassVar[ThreadPoolExecutor] = ThreadPoolExecutor(
+        2, thread_name_prefix="TCP-Executor"
+    )
+    _client: ClassVar[TCPClient]
+
+    @classmethod
+    def warmup(cls) -> None:
+        """Pre-start threads and sockets to avoid catching them in checks for thread and
+        fd leaks
+        """
+        ex = cls._executor
+        while len(ex._threads) < ex._max_workers:
+            ex._adjust_thread_count()
+        cls._get_client()
+
+    @classmethod
+    def _get_client(cls):
+        if not hasattr(cls, "_client"):
+            resolver = netutil.ExecutorResolver(
```

Comment:
The ExecutorResolver is deprecated in tornado 5+

https://github.com/tornadoweb/tornado/blob/e18ea03f87c95dc2f39ca94188fd32016664e3ff/tornado/netutil.py#L423-L425

can we just use a plain TCPClient instead?

Reply:
Sure, but I'd rather leave it to a separate PR